# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformFontList.cpp
# Commit: 646176bf3689
# Full Hash: 646176bf368922cc42eb1f40ff6a28c91e54bb34
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-12-26 21:33:51
# Regressor Bug: 1669855
# File Overlap Count: 1
# Description:
#   Bug 1669855 - Support triggering the background font-info loader immediately if it is currently deferred, and use this to implement InitOtherFamilyNames for the shared font list. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96395
# ==============================================================================

diff -r 87186903dd5f -r 646176bf3689 gfx/thebes/gfxPlatformFontList.cpp
--- a/gfx/thebes/gfxPlatformFontList.cpp	Sat Dec 26 17:11:52 2020 +0000
+++ b/gfx/thebes/gfxPlatformFontList.cpp	Sat Dec 26 17:12:04 2020 +0000
@@ -2363,14 +2363,11 @@
 
     auto list = SharedFontList();
     if (list) {
-      for (auto& f : mozilla::Range<fontlist::Family>(list->Families(),
-                                                      list->NumFamilies())) {
-        ReadFaceNamesForFamily(&f, false);
-        TimeDuration elapsed = TimeStamp::Now() - start;
-        if (elapsed.ToMilliseconds() > OTHERNAMES_TIMEOUT) {
-          timedOut = true;
-          break;
-        }
+      // If the gfxFontInfoLoader task is not yet running, kick it off now so
+      // that it will load remaining names etc as soon as idle time permits.
+      if (mState == stateInitial || mState == stateTimerOnDelay) {
+        StartLoader(0);
+        timedOut = true;
       }
     } else {
       for (auto iter = mFontFamilies.Iter(); !iter.Done(); iter.Next()) {
